core/api/src/main/scala/org/virtuslab/ideprobe/protocol/ModuleRef.scala

```diff
@@ -1,9 +1,17 @@
 package org.virtuslab.ideprobe.protocol
 
+import org.virtuslab.ideprobe.ConfigFormat
+import pureconfig.{ConfigReader, ConfigWriter}
+import pureconfig.generic.semiauto.{deriveReader, deriveWriter}
+
 final case class ModuleRef(name: String, project: ProjectRef = ProjectRef.Default)
 
-object ModuleRef {
+object ModuleRef extends ConfigFormat {
   def apply(name: String, project: String): ModuleRef = {
     ModuleRef(name, ProjectRef(project))
   }
+
+  implicit val moduleRefReader: ConfigReader[ModuleRef] = deriveReader[ModuleRef]
```

Comment:
I think here is deriveConvert that derives ConfigConvert which does both reading and writing